normandy/base/middleware.py

```diff
@@ -117,3 +122,150 @@ def process_request(self, request):
                 }
             )
         return response
+
+
+class OIDCEndpointRequestError(Exception):
+    """Happens when the server-to-server communication with the OIDC endpoint
+    succeeds but the OIDC endpoints responds with a status code > 200 and less
+    than 500."""
+
+
+class OIDCAccessTokenAuthorizationMiddleware:
+    """Middleware that looks for 'Authorization: Bearer access_token' headers in
+    the request and uses this access token to use an OIDC provider to turn that into
+    a user profile. If that works, the request becomes signed in as this user.
+
+    Note, it only signs in the user's request. Not session. Since this is not
+    an authentication backend, there is no 'Set-Cookie' header returned in the
+    response.
+
+    It's admittedly confusing to mix the terms authentication and authorization.
+    Arguably the client has already *authenticated* and basically converted her
+    password for an access token. Here in the context of Normandy we kill two
+    birds with one stone: check that access token and use the user profile that
+    comes out to sign the user into the rest of the Django request.
```

Comment:
I don't think this paragraph adds much to a reader who wasn't a part of our debate. I'd like to just take it out entirely.